quill-core/src/main/scala/io/getquill/monad/ScalaFutureIOMonad.scala

```diff
@@ -0,0 +1,30 @@
+package io.getquill.monad
+
+import scala.util.Failure
+import scala.util.Success
+import io.getquill.context.Context
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
+
+trait ScalaFutureIOMonad extends IOMonad {
+  this: Context[_, _] =>
+
+  type Result[T] = Future[T]
+
+  def unsafePerformIO[T](io: IO[T, _], transactional: Boolean = false)(implicit ec: ExecutionContext): Result[T] =
+    io match {
+      case FromTry(v) => Future.fromTry(v)
+      case Run(f)     => f()
+      case Sequence(in, cbfIOToResult, cbfResultToValue) =>
+        val builder = cbfIOToResult()
+        in.foreach(builder += unsafePerformIO(_))
+        Future.sequence(builder.result)(cbfResultToValue, ec)
```

Comment:
Will this leads `postgres-async` and `mysql-async` to run multi  query at the same time ?

Reply:
@jilen yes. Do you ask that because of the concurrency issue with parallel queries under a transaction? I think that's a driver bug. For instance, ndbc handles this scenario without problems.

Reply:
@fwbrasil  Yes. And if it not support `sequence` we could  point it out somewhere.

Reply:
@jilen I think that's a separate issue. The same happens with `Future.sequence`. Could you make sure that we have an issue open for it? If the driver doesn't fix it, we can workaround on the quill side by having a synchronization point for transactions.

Reply:
@fwbrasil  I'll open an issue while this pr  merged